### PR TITLE
bump poetry and deps

### PR DESCRIPTION
**Pull request by KotlinIsland**

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -3,27 +3,27 @@ name: check
...
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - name: Run image
-        uses: abatilo/actions-poetry@v2.0.0
+        uses: abatilo/actions-poetry@v2
```
Comment (DetachHead) [inline .github/workflows/check.yaml]: i think it's better to explicitly bump these because actions sometimes have breaking changes

Comment (KotlinIsland) [reply .github/workflows/check.yaml]: Okay

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -39,16 +39,16 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
         with:
           python-version: "3.7"
       - name: Run image
-        uses: abatilo/actions-poetry@v2.0.0
+        uses: abatilo/actions-poetry@v2
```
Comment (DetachHead) [inline .github/workflows/check.yaml]: this one as well? maybe the version can be set to a variable?

Comment (KotlinIsland) [reply .github/workflows/check.yaml]: I'll update it, but I haven't seen that pattern used anywhere else so I don't really see any value in the added comploxity.

Comment (DetachHead) [reply .github/workflows/check.yaml]: maybe its fine as is then, i just remember an action i was using once broke something but maybe thats not common enough to justify pinning them. up to u

Comment (KotlinIsland) [reply .github/workflows/check.yaml]: I think it's good to pin them, but using a variable seems like unknown complexity

**Pull request merged**